Migrate-AvatarIcon

thebinij opened the pull request:
## Explanation
Closes #18883  
<!--
Thanks for the pull request. Take a moment to answer these questions so that reviewers have the information they need to properly understand your changes:

* What is the current state of things and why does it need to change?
* What is the solution your changes offer and how does it work?

Are there any issues, Slack conversations, Zendesk issues, user stories, etc. reviewers should consult to understand this pull request better? For instance:

* Fixes #12345
* See: #67890
-->

<!-- How does it look now? Drag your file(s) below this line: -->

## Manual Testing Steps

No functional changes

<!--
How should reviewers and QA manually test your changes? For instance:

- Go to this screen
- Do this
- Then do this
-->

## Pre-merge author checklist

- [ ] I've clearly explained:
  - [x] What problem this PR is solving
  - [ ] How this problem was solved
  - [ ] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [x] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

micaelae committed:
Show Bridge button in TokenOverview component (#18630)

* Show Bridge button in TokenOverview component

* Hide Swap button in token overview page when network is not supported

micaelae committed:
Show Bridge button in TokenOverview component (#18630)

* Show Bridge button in TokenOverview component

* Hide Swap button in token overview page when network is not supported

jpuri committed:
Fixing issue with gasLimit in transaction being 0 (#18682)

Gudahtt committed:
Update outdated browser versions (#18721)

The range of browsers considered to be outdated has been updated to
match our latest browser support policy, which is ~2 years on Chrome,
and the most recent end-of-life extended support release on Firefox.

Comments have been added to help explain how these versions were chosen
as well.

jpuri committed:
Enable editing L2 gas on optimism (#18217)

danjm committed:
Fix mv3 beta build (#18690)

* Build beta with mv3 enabled

* Ensure firefox manifest is an mv2 version

* Revert "Ensure firefox manifest is an mv2 version"

This reverts commit fed74792b0fec33c3a85f2229eb560559d37afe5.

* Only create beta builds for the chrome platform

* Stop linting firefox for beta

thebinij committed:
Migrate-AvatarIcon

thebinij committed:
Merge branch 'develop' into #18883-Migrate-AvatarIcon

thebinij committed:
Merge branch 'MetaMask:develop' into #18883-Migrate-AvatarIcon

thebinij committed:
fix depcheck issue

thebinij committed:
fix lint

thebinij committed:
Merge branch 'develop' into #18883-Migrate-AvatarIcon

pedronfigueiredo committed:
Label PRs based on the labels of the associated issue (#17603)

* Implement CI to copy issue labels over to PRs

* wip

* wip

* wip

* wip

* wip

* wip

* wip

* wip

* wip

* clean up

* clean up

vinistevam committed:
Fix fail to reject multiple approval requests (#19050)

PrgrmrHarshShukla committed:
Update slider.component.js (#19035)

dhruvv173 committed:
Changes to confirm-data and confirm-hexdata (#19041)

NidhiKJha committed:
UX Multichain: Added extra margin for native token (#18988)

* added extra margin for native token

* fixed lint errors

* added marginTop via css

* updated test

* added boxprops and margin bottom for tab component

* updated snapshot

* update asset list component

legobeat committed:
devDeps: storybook-dark-mode@1.1.2->2.1.1 (#18866)

* devDeps: @storybook/*->6.5.16

* devDeps: storybook-dark-mode@1.1.2->2.1.1

* add lavamoat build policy override

PrgrmrHarshShukla committed:
Working on some lint issues in 2 files. (#18633)

* Addressed some lint issues.

* Worked on the mentioned lint issue.

* Worked on the CRLF lint issues.

---------

Co-authored-by: Brad Decker <bhdecker84@gmail.com>
Co-authored-by: George Marshall <george.marshall@consensys.net>

darkwing committed:
UX: Multichain: Account Details Fixes (#18999)

* Fix account details editable label

* Fix font-weight of the password label

* Remove duplicate error

* Hide password warning on first screen after bad password, autofocus password field

* Fix jest failure

* Restore className

zone-live committed:
[MMI] adds mmi logic in home view (#18960)

* adds mmi logic to home container

* adds code fences to home component

* adds code fences to home component

* lint fix

* lint fix

* updates fences and runs prettier

* version bump

* clean up code fence

PrgrmrHarshShukla committed:
Part of #17670 & #18714: Replace Typography with Text component: nft-details.js (#18771)

* Update nft-details.js

* Update nft-details.test.js.snap

* Update nft-details.js

* Update ui/components/app/nft-details/nft-details.js

Co-authored-by: George Marshall <georgewrmarshall@gmail.com>

* Update ui/components/app/nft-details/nft-details.js

Co-authored-by: George Marshall <georgewrmarshall@gmail.com>

* Update ui/components/app/nft-details/nft-details.js

Co-authored-by: George Marshall <georgewrmarshall@gmail.com>

* Update ui/components/app/nft-details/nft-details.js

Co-authored-by: George Marshall <georgewrmarshall@gmail.com>

* Update ui/components/app/nft-details/nft-details.js

Co-authored-by: George Marshall <georgewrmarshall@gmail.com>

* Update ui/components/app/nft-details/nft-details.js

Co-authored-by: George Marshall <georgewrmarshall@gmail.com>

* Update ui/components/app/nft-details/nft-details.js

Co-authored-by: George Marshall <georgewrmarshall@gmail.com>

* Update ui/components/app/nft-details/nft-details.js

Co-authored-by: George Marshall <georgewrmarshall@gmail.com>

* Update ui/components/app/nft-details/nft-details.js

Co-authored-by: George Marshall <georgewrmarshall@gmail.com>

* Update ui/components/app/nft-details/nft-details.js

Co-authored-by: George Marshall <georgewrmarshall@gmail.com>

* Update ui/components/app/nft-details/nft-details.js

Co-authored-by: George Marshall <georgewrmarshall@gmail.com>

* Update ui/components/app/nft-details/nft-details.js

Co-authored-by: George Marshall <georgewrmarshall@gmail.com>

* Update ui/components/app/nft-details/nft-details.js

Co-authored-by: George Marshall <georgewrmarshall@gmail.com>

* Update ui/components/app/nft-details/nft-details.js

Co-authored-by: George Marshall <georgewrmarshall@gmail.com>

* Update nft-details.test.js.snap

---------

Co-authored-by: legobeat <109787230+legobeat@users.noreply.github.com>
Co-authored-by: Brad Decker <bhdecker84@gmail.com>
Co-authored-by: George Marshall <georgewrmarshall@gmail.com>
Co-authored-by: George Marshall <george.marshall@consensys.net>

darkwing committed:
Fix #18916 - UX: Multichain: Network picker design updates (#18962)

* Remove border radius from network menu list

* Provide background color for test network avatars

* Make button size large

* Update ui/components/multichain/network-list-item/network-list-item.js

Co-authored-by: Nidhi Kumari <nidhi.kumari@consensys.net>

* Fix lint

---------

Co-authored-by: Nidhi Kumari <nidhi.kumari@consensys.net>

darkwing committed:
UX: Multichain: Ensure network picker renders properly for long and short names (#18974)

* UX: Multichain: Ensure network picker renders properly for long and short network names

* Updatte ui/components/multichain/app-header/app-header.js

Co-authored-by: George Marshall <george.marshall@consensys.net>

* Update ui/components/multichain/app-header/app-header.js

Co-authored-by: George Marshall <george.marshall@consensys.net>

* Update ui/components/multichain/app-header/app-header.js

Co-authored-by: George Marshall <george.marshall@consensys.net>

* Update ui/components/multichain/app-header/app-header.js

Co-authored-by: George Marshall <george.marshall@consensys.net>

* Update ui/components/multichain/app-header/app-header.js

Co-authored-by: George Marshall <george.marshall@consensys.net>

* Update ui/components/multichain/app-header/app-header.scss

Co-authored-by: George Marshall <george.marshall@consensys.net>

* Update ui/components/multichain/app-header/app-header.scss

Co-authored-by: George Marshall <george.marshall@consensys.net>

* Fix jest issue

---------

Co-authored-by: George Marshall <george.marshall@consensys.net>

legobeat committed:
devDeps: react-devtools@4.16.0->4.27.6 (#18869)

dhruvv173 committed:
Changes to custody-labels.js (#19038)

darkwing committed:
UX: Multichain: Disable network and account picker when necessary (#18909)

darkwing committed:
UX: Multichain: Update deprecated variables (#19058)

darkwing committed:
Fix #19059 - Show network picker when locked (#19063)

darkwing committed:
Fix #19060 - Ensure there's multiple accounts before focusing search (#19062)

dominikrudzki committed:
Fix popover scroll button WhatsNewPopup hiding (#19017)

Co-authored-by: David Walsh <davidwalsh83@gmail.com>

NidhiKJha committed:
UX Multichain: updated margin top for assets tab (#19071)

* updated margin top for assets tab

* remove changes from yml file

* remove changes from yml file

NidhiKJha committed:
Replaced all fa-icon-circle with INFO icon (#17539)

* replace all fa-icon-circle with INFO icon

* updated classnames

* updated classnames

* updated snapshots

* updated colors

* resolved errors

* fixed relative import

* fixed lint errors

* added story for alerts tab

* update snapshot

* updated info-circle

* updated enum for iconName

* removed classnames

* updated iconName in settings

* fixed lint errors and snapshots

Gudahtt committed:
Set network status to "unknown" when ID is invalid (#19068)

We now set the network status to "unknown" rather than "unavailable"
when the network ID is invalid. This better reflects what we know when
this happens, and it makes the network controller better aligned with
the core network controller.

This was accomplished by using a regular error for the network ID
assertion rather than using `assert` directly. `assert` would throw an
error with a `code` property, which resutled in us treating it like an
RPC error.

This isn't tested, but it was found in the course of porting unit tests
from core to extension. It will be covered by these tests, which will
be added in the next PR.

This change should have no functional impact because we treat these two
network statuses as equivalent. The distinction between unknown and
unavailable is useful only for debugging.

pedronfigueiredo committed:
remove pre commit hook for branch names (#19077)

* remove pre commit hook for branch names

* bail with ticket number 0 detected

ritave committed:
Made builds.yml variable errors more helpful (#19066)

georgewrmarshall committed:
Some style, accessibility and sematic html updates to modal sub components (#19034)

georgewrmarshall committed:
Adding ModalFocus component (#18979)

georgewrmarshall commented on yarn.lock:
```diff
@@ -11512,9 +11512,9 @@ __metadata:
+  checksum: a5f7681c860a29736f29350ebd81041c40b6aa7b2f94c50ed27284a0507e46dc79536dcfc05432504cfc80a0bf2070e4ad6fa704a9c0f3f32d47bed9059e98c2
   languageName: node
   linkType: hard
 
```
Please revert this file

georgewrmarshall reviewed (commented):
Some of the comments in the other PRs relate to this one so I will do a full review after the others have been addressed

dhruvv173 committed:
Changes to edit-gas-display.component.js (#19037)

Co-authored-by: Brad Decker <bhdecker84@gmail.com>

PrgrmrHarshShukla committed:
Part of #17670 & #18714: Replace Typography with Text component in token-details-page.js (#18741)

* Worked on issue #17670 & #18714 in file token-details-page.js in file

* Worked on issue #17670 & #18714 in file token-details-page.js in file

* Update token-details-page.js

vinistevam committed:
Adopt ApprovalType from core (#18567)

legobeat committed:
builds.yml: add env var EDITOR_URL (#19074)

mcmire committed:
Re-enable tests for subscription-based RPC methods (#18994)

NetworkController doesn't handle `eth_subscribe` or `eth_unsubscribe`
specially, but as it's supported by Infura, we want to make sure we
exercise these RPC methods in the network client tests, even if it is
just to ensure that they get passed through to the network.

We had tests for these RPC methods, but they were commented out in
c095b1accd00c8123f954633bf396ddea7b95eff when the network client code
was extracted to a separate file. At the time we were considering adding
subscription- and filter-based middleware to NetworkController, and so
we commented out the tests for `eth_subscribe` and `eth_unsubscribe`
temporarily until we could rewrite them in a way that would exercise the
new middleware. We reverted that change in
bd23a490133693a464c2f4ab1976181378a9f7bf, which meant that we could
restore the existing tests, but it appears that this task was not caught
during review. This commit takes care of restoring them.

Co-authored-by: legobeat <109787230+legobeat@users.noreply.github.com>

NidhiKJha committed:
removed mmi fencing from copy icon in header (#19087)

* removed mmi fencing from copy icon

* updated snapshot

Gudahtt committed:
Only recognize "blocked" status for built-in networks (#19069)

The detection of the Infura "blocked" status has been updated to apply
only to built-in networks. The message we show to users in this state
is meant only for Infura; we don't want to show it for third-party RPC
APIs that happen to use the same error response.

This brings the network controller further in alignment with the core
network controller.

This isn't tested, but it was found in the course of porting unit tests
from core to extension. It will be covered by these tests, which will
be added in the next PR.

NEllusion committed:
Fix script to build migrations by having it use the `.ts` template (#18878)

* Fix script to build migrations

* Validate version is a number

OGPoyraz committed:
feat: Refactor Transaction Confirmation selector (#18796)

OGPoyraz committed:
Add `getCurrentChainId` argument to `SignatureController` (#19078)

OGPoyraz committed:
fix: use approvals selector fn in permission selectors (#19095)

bowensanders committed:
[FLASK] Create E2E test for snap_GetEntropy (#18998)

thebinij committed:
revert yarn.lock changes

thebinij committed:
rebasing

thebinij committed:
replace old TEXT_ALIGN
